suprb/solution/mixing_model.py

```diff
@@ -37,19 +107,22 @@ def __call__(self, X: np.ndarray, subpopulation: list[Rule], cache=False) -> np.
                     continue
                 local_pred[i][matches[i]] = rule.predict(X[matches[i]])
 
-        taus = (1 / errors) * experiences
+        return local_pred, matches
 
-        # Stack all local predictions and sum them weighted with tau
-        pred = np.sum(local_pred * taus[:, None], axis=0)
+    def _get_taus(self, subpopulation: list[Rule], dim: int):
+        # Get errors and experience of all rules in subpopulation
+        experiences = self.experience_calculation(subpopulation, dim)
+        errors = np.array([rule.error_ for rule in subpopulation])
 
+        return (1 / errors) * (experiences * self.experience_weight)
+
+    def _get_tau_sum(self, subpopulation: list[Rule], matches: list[Rule], taus: list[int]):
         # Sum all taus
-        local_taus = np.zeros((len(subpopulation), input_size))
+        local_taus = np.zeros((len(subpopulation), self.input_size))
         for i in range(len(subpopulation)):
             local_taus[i][matches[i]] = taus[i]
 
         tau_sum = np.sum(local_taus, axis=0)
         tau_sum[tau_sum == 0] = 1  # Needed
```

Comment:
can you add a comment why this is needed?

Reply:
Done, but please check if that reason makes sense, since I didn't implement that part of the logic